CORE-1412 Consul Provider

athompson-r7 opened the pull request:
This Pull Request adds the Consul Provider.

The useage looks something like this:

``` ruby
instances = Cluster::Discovery.discover(
  'consul',
  consul_url: 'http://my.consul.cluster:8500',
  consul_service: 'redis',
  leader: true,
  tags: 'master')
instances.map(&:Address)
```

athompson-r7 committed:
CORE-1412 Update dependencies

athompson-r7 committed:
CORE-1412 Consul discovery provider

This commit adds a consul discovery provider.  This requires an existing
Consul cluster, a service name, and optionally tags and leader state.

athompson-r7 committed:
CORE-1412 Fixes discovery by tag

This commit fixes discovery by tag aswell as requesting the `:leader`
not returning an Array.

athompson-r7 committed:
CORE-1412 Fixes consul proxy method

This commit fixes the consul proxy method and adds docs and tests.

athompson-r7 committed:
CORE-1412 Load fixtures

This commit adds fixture loaders and the existing cassettes.

athompson-r7 committed:
Bumping #minor version

athompson-r7 committed:
CORE-1412 Fix copy/paste error in the README

fmitchell-r7 commented on spec/lib/cluster/discovery/consul_spec.rb:
```diff
@@ -0,0 +1,119 @@
+  describe '.discover', vcr: true do
+    before(:each) do
+      @consul = Cluster::Discovery::Consul.new(
+        consul_url: 'http://172.28.128.104:8500')
```
Should this IP address by `ENV['TEST_CONSUL_HOST']` so it makes up with the Rakefile?

fmitchell-r7 commented on spec/lib/cluster/discovery_spec.rb:
```diff
@@ -7,5 +7,17 @@
+      it 'discovery with consul' do
+        consul = Cluster::Discovery.discover(
+          'consul',
+          consul_url: 'http://172.28.128.104:8500',
```
Should this IP address by `ENV['TEST_CONSUL_HOST']` so it makes up with the Rakefile?

athompson-r7 committed:
CORE-1412 Make Consul host setting consistent

This commit tries to make the consul host setting consistent between the
specs and the Rakefile.  This _really_ only has any effect if the
request hasn't been recored.

jmanero-r7 commented:
:+1:
